File: base/src/main/vsdk/toolkit/processing/polyhedralBoundedSolidOperators/_PolyhedralBoundedSolidSetOperatorVertexVertex.java
//= References:                                                             =
//= [MANT1988] Mantyla Martti. "An Introduction To Solid Modeling",         =
//=     Computer Science Press, 1988.                                       =

package vsdk.toolkit.processing.polyhedralBoundedSolidOperators;

import vsdk.toolkit.environment.geometry.volume.polyhedralBoundedSolid.nodes._PolyhedralBoundedSolidVertex;

/**
Stores one vertex/vertex coincidence from the `sonvv` set introduced by
program [MANT1988].15.1.
*/
public class _PolyhedralBoundedSolidSetOperatorVertexVertex
    extends PolyhedralBoundedSolidOperator
{
    public _PolyhedralBoundedSolidVertex va;
    public _PolyhedralBoundedSolidVertex vb;

    public String toString()
    {
        String msg = "(" + va + ") / (" + vb + "}";
        return msg;
    }
}

File: typescript/base/src/main/vsdk/toolkit/environment/geometry/geometricProcessing/polyhedralBoundedSolidOperators/booleans/classification/_PolyhedralBoundedSolidSetOperatorVertexVertex.ts
import { _PolyhedralBoundedSolidOperator } from "../../_PolyhedralBoundedSolidOperator.js";
import type { _PolyhedralBoundedSolidVertex } from "../../../../volume/polyhedralBoundedSolid/nodes/_PolyhedralBoundedSolidVertex.js";
export class _PolyhedralBoundedSolidSetOperatorVertexVertex extends _PolyhedralBoundedSolidOperator {
    public va: _PolyhedralBoundedSolidVertex | null = null;
    public vb: _PolyhedralBoundedSolidVertex | null = null;
    public override toString(): string {
        return `(${this.va}) / (${this.vb}}`;
    }
}
